pkg/apk/impl/fs/memfs.go

```diff
@@ -17,77 +17,156 @@ package fs
 import (
 	"archive/tar"
 	"bytes"
-	"encoding/binary"
 	"errors"
 	"fmt"
 	"io"
 	"io/fs"
 	"os"
 	"path/filepath"
 	"strings"
+	"time"
 
-	"github.com/blang/vfs"
-	origMemfs "github.com/blang/vfs/memfs"
 	"golang.org/x/sys/unix"
 )
 
-const (
-	linkFlagSymlink  = 's'
-	linkFlagHardlink = 'h'
-)
-
-func NewMemFS() FullFS {
-	return &memFS{origMemfs.Create(), map[string]*ownership{}}
+type memFS struct {
+	tree *node
 }
 
-type ownership struct {
-	uid, gid int
-	perms    fs.FileMode
-}
-type memFS struct {
-	*origMemfs.MemFS
-	perms map[string]*ownership // used only for overrides
+func NewMemFS() *memFS {
```

Comment:
nit: It might be nice to return the interface type here, if you don't want to export the `memFS` type. It would make it more clear in this context what interface `memFS` is intending to implement, and it also makes it easier to traverse expected behaviors in docs/IDEs for folks calling `NewMemFS()`.

Reply:
Yeah, that is a good idea. I went back and forth on it and just defaulted, but I think you are correct.

Reply:
Done